doc/reference/networking/lwm2m.rst

```diff
@@ -11,8 +11,281 @@ Overview
+| 15 | Timezone          | RW  | Single   | Optional  | String  |
++----+-------------------+-----+----------+-----------+---------+
+| 16 | Supported Binding | R   | Single   | Mandatory | String  |
++----+-------------------+-----+----------+-----------+---------+
+| 17 | Device Type       | R   | Single   | Optional  | String  |
++----+-------------------+-----+----------+-----------+---------+
+| 18 | Hardware Version  | R   | Single   | Optional  | String  |
++----+-------------------+-----+----------+-----------+---------+
+| 19 | Software Version  | R   | Single   | Optional  | String  |
++----+-------------------+-----+----------+-----------+---------+
+| 20 | Battery Status    | R   | Single   | Optional  | Integer |
+|    |                   |     |          |           | 0-6     |
++----+-------------------+-----+----------+-----------+---------+
+| 21 | Memory Total (Kb) | R   | Single   | Optional  | Integer |
++----+-------------------+-----+----------+-----------+---------+
+| 22 | ExtDevInfo        | R   | Multiple | Optional  | ObjLink |
++----+-------------------+-----+----------+-----------+---------+
+
+The server could query the `Device Manufacturer` for `Device` object instance 0
+(the default and only instance) by sending a `READ 3/0/0` operation to the
+client.
+
+The full list of registered objects and resource IDs can be
+found in the [LwM2M registry](http://www.openmobilealliance.org/wp/OMNA/LwM2M/LwM2MRegistry.html).
+
+Zephyr's LwM2M library lives in the `subsys/net/lib/lwm2m directory`, with a
+client sample in `samples/net/lwm2m_client`.  For more information about the
+provided sample see: ref:`lwm2m-client-sample`  The sample can be configured
+to use normal unsecure network sockets or sockets secured via DTLS.
+
+The Zephyr LwM2M library implements the following items:
+
+* engine to process networking events and core functions
+* RD client which performs BOOTSTRAP and REGISTRATION functions
+* TLV, JSON and plain text formatting functions
+* LwM2M Technical Specification Enabler objects such as Security, Server,
+  Device, Firmware Update, etc.
+* Extended IPSO objects such as Light Control, Temperature Sensor and Timer
+
+The library currently implements up to [LwM2M specification 1.0.2](http://openmobilealliance.org/release/LightweightM2M/V1_0_2-20180209-A/OMA-TS-LightweightM2M-V1_0_2-20180209-A.pdf).
+
+For more information about LwM2M visit [OMASpecWorks LwM2M](https://www.omaspecworks.org/what-is-oma-specworks/iot/lightweight-m2m-lwm2m/).
+
+Sample usage
+************
+
+To use the LwM2M library, start by creating an LwM2M client context
+`struct lwm2m_ctx` structure:
+
+.. code-block:: c
+
+   /* LwM2M client context */
+   static struct lwm2m_ctx client;
+
+Create callback functions for LwM2M resources that you wish to have actions
+for:
+
+.. code-block:: c
+
+   static int device_reboot_cb(u16_t obj_inst_id)
+   {
+       LOG_INF("Device rebooting.");
```

Comment:
Minor thing but if possible we could follow zephyr style guide here, so there should be eight spaces (one tab) from start of the line. Same comment to all the code examples. If it is too much work, then what we have here is fine too.